launchable/commands/split_subset.py

```diff
@@ -180,18 +250,76 @@ def run(self):
                         err=True)
                     return
 
-            if len(output) == 0:
-                click.echo(click.style("Error: no tests found in this subset id.", 'yellow'), err=True)
-                return
+        def _write_split_by_groups_group_names(self, subset_group_names: List[str], rest_group_names: List[str]):
+            if is_output_exclusion_rules:
+                subset_group_names, rest_group_names = rest_group_names, subset_group_names
+
+            if len(subset_group_names) > 0:
+                with open("{}/{}".format(split_by_groups_output_dir, SPLIT_BY_GROUP_SUBSET_GROUPS_FILE_NAME),
+                          "w+", encoding="utf-8") as f:
+                    f.write("\n".join(subset_group_names))
+
+            if is_split_by_groups_with_rest and len(rest_group_names) > 0:
+                with open("{}/{}".format(split_by_groups_output_dir, SPLIT_BY_GROUP_REST_GROUPS_FILE_NAME),
+                          "w+", encoding="utf-8") as f:
+                    f.write("\n".join(rest_group_names))
+
+        def split_by_group_namess(self):
+            try:
+                res = client.request("POST", "{}/split-by-groups".format(subset_id))
+                res.raise_for_status()
+
+                is_observation = res.json().get("isObservation", False)
+                split_groups = res.json().get("splitGroups", {})
+
+                subset_group_names = []
+                rest_group_names = []
+
+                for group in split_groups:
+                    group_name = group.get("groupName", "")
+                    subset = group.get("subset", [])
+                    rests = group.get("rest", [])
+
+                    if is_observation:
+                        subset, rests = subset + rests, []
+
+                    if is_output_exclusion_rules:
+                        for g in split_groups:
+                            if g.get("groupName", "") != group_name:
```

Comment:
[question]
Is it correct that you do `for g in split_groups` inside of `for group in split_groups`?

Reply:
I want to get all subsets and rests of other groups. Is there any good way to get it without loops?

Reply:
If that is the case, then this loop is fine.
Another way is to write in list comprehension, which tends to be faster than usual for loop in Python, but not mandatory.

Maybe something like this (I didn't tested):

```python
rests += [g.get("subset", []) + g.get("rest", []) for g in split_groups if g.get("groupName", "") != group_name]
```

Reply:
Anyway you don't have to fix here if you feel unnecessary.

Reply:
one line is a little difficult to read (in my case), so I'll keep it. Thanks